src/main/java/roomescape/service/ReservationTimeService.java

```diff
@@ -0,0 +1,75 @@
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import roomescape.domain.Reservation;
+import roomescape.domain.ReservationTime;
+import roomescape.dto.AvailableReservationTimeResponse;
+import roomescape.dto.ReservationTimeRequest;
+import roomescape.dto.ReservationTimeResponse;
+import roomescape.domain.ReservationRepository;
+import roomescape.domain.ReservationTimeRepository;
+
+@Service
+@Transactional(readOnly = true)
+public class ReservationTimeService {
+
+    private final ReservationRepository reservationRepository;
+    private final ReservationTimeRepository reservationTimeRepository;
+
+    public ReservationTimeService(ReservationRepository reservationRepository,
+                                  ReservationTimeRepository reservationTimeRepository) {
+        this.reservationRepository = reservationRepository;
+        this.reservationTimeRepository = reservationTimeRepository;
+    }
+
+    public List<ReservationTimeResponse> getAllReservationTimes() {
+        List<ReservationTime> reservationTimes = reservationTimeRepository.findAll();
+
+        return reservationTimes.stream()
+                .map(ReservationTimeResponse::from)
+                .toList();
+    }
+
+    @Transactional
+    public ReservationTimeResponse addReservationTime(ReservationTimeRequest reservationTimeRequest) {
+        ReservationTime reservationTime = reservationTimeRequest.toReservationTime();
+
+        if (reservationTimeRepository.existsByStartAt(reservationTime.getStartAt())) {
+            throw new IllegalArgumentException("해당 시간은 이미 존재합니다.");
+        }
+
+        ReservationTime savedReservationTime = reservationTimeRepository.save(reservationTime);
+
+        return ReservationTimeResponse.from(savedReservationTime);
+    }
+
+    @Transactional
+    public void deleteReservationTimeById(Long id) {
+        if (!reservationTimeRepository.existsById(id)) {
+            throw new NoSuchElementException("해당 id의 시간이 존재하지 않습니다.");
+        }
+
+        if (reservationRepository.existsByTimeId(id)) {
+            throw new IllegalArgumentException("해당 시간을 사용하는 예약이 존재합니다.");
+        }
+
+        reservationTimeRepository.deleteById(id);
+    }
+
+    public List<AvailableReservationTimeResponse> getAvailableReservationTimes(LocalDate date, Long themeId) {
+        List<ReservationTime> reservationTimes = reservationTimeRepository.findAll();
+        List<Reservation> reservations = reservationRepository.findByDateAndThemeId(date, themeId);
+
+        return reservationTimes.stream()
+                .map(reservationTime -> new AvailableReservationTimeResponse(
+                        reservationTime.getId(),
+                        reservationTime.getStartAt(),
+                        reservations.stream()
+                                .anyMatch(reservation -> reservationTime.equals(reservation.getTime())))
+                ).toList();
```

Comment:
```suggestion
       return reservationTimes.stream()
                .map(reservationTime -> {
                    boolean available = reservations.stream()
                        .anyMatch(reservation -> reservationTime.equals(reservation.getTime()));

                    return new AvailableReservationTimeResponse(
                        reservationTime.getId(),
                        reservationTime.getStartAt(), 
                        available
                    );
                }).toList();
```

한번에 처리하기보다는 적절히 변수분리를 해주는게 가독성이 더 좋을 수 있을것 같아요!

Reply:
메서드로 분리해서 개선했습니다!

```java
public List<AvailableReservationTimeResponse> getAvailableReservationTimes(LocalDate date, Long themeId) {
    List<ReservationTime> reservationTimes = reservationTimeRepository.findAll();
    List<Reservation> reservations = reservationRepository.findByDateAndThemeId(date, themeId);

    return reservationTimes.stream()
            .map(reservationTime -> new AvailableReservationTimeResponse(
                    reservationTime.getId(),
                    reservationTime.getStartAt(),
                    isAlreadyBooked(reservationTime, reservations))
            ).toList();
}

private boolean isAlreadyBooked(ReservationTime reservationTime, List<Reservation> reservations) {
    return reservations.stream()
            .anyMatch(reservation -> reservationTime.equals(reservation.getTime()));
}
```

Reply:
죄송합니다. 이 부분은 query 2개를 합쳐서 수정했었습니다.